### PR TITLE
Update RELEASES.md for 1.25.0

### PR DESCRIPTION
**Pull request by XAMPPRocky**
[rendered](https://github.com/Aaronepower/rust/blob/master/RELEASES.md)

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,100 @@
...
+- [Added `i586-unknown-linux-musl` target][47282]
+
+Libraries
+---------
+- [Impl Send for `process::Command` on Unix.][47760]
+- [Impl PartialEq and Eq for `ParseCharError`.][47790]
+- [`UnsafeCell::into_inner` is now safe.][47204]
+- [Implement libstd for CloudABI.][47268]
+- [`Float::{from_bits, to_bits}` is now available in libcore.][46931]
+- [renamed `ptr::Shared` to `ptr::NonNull`.][46952]
```
Comment (steveklabnik) [inline RELEASES.md]: this isn't stable yet, so probably shouldn't be in here, right?

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,100 @@
...
+----
+
+Compatibility Notes
+-------------------
+- [Deprecated `net::lookup_host`.][47510]
+- [`rustdoc` has switched to pulldown as the default markdown renderer.][47398]
+- The borrow checker was sometimes incorrectly permitting overlapping borrows
+  around indexing operations (see #47349). This has been fixed (which also
+  enabled some correct code that used to cause errors (e.g. #33903 and #46095).
+- [Removed deprecated unstable attribute `#[simd]`.][47251]
```
Comment (steveklabnik) [inline RELEASES.md]: same here

File: RELEASES.md (hunk 1 above)
Comment (Aaronepower) [reply RELEASES.md]: @steveklabnik Looking at the beta docs and the prs this mentioned in. [It's stable this release](https://doc.rust-lang.org/beta/std/ptr/struct.NonNull.html) unless there was a revert pr I didn't see.

File: RELEASES.md (hunk 2 above)
Comment (Aaronepower) [reply RELEASES.md]: @steveklabnik @Mark-Simulacrum gave it the `relnotes` tag so I included it.

File: RELEASES.md (hunk 1 above)
Comment (steveklabnik) [reply RELEASES.md]: yeah, I would make this link to the stabilization of `NonNull` in that case.

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,102 @@
+Version 1.25.0 (2018-03-29)
+==========================
+
+Language
+--------
+- [Stabilised `#[repr(align(x))]`.][47006] [RFC 1358]
+- [You can now use nested groups of imports.][47948]
+  eg. `use std::{fs::File, io::Read, path::{Path, PathBuf}};`
```
Comment (Mark-Simulacrum) [inline RELEASES.md]: We should probably use `e.g.` instead of `eg.` throughout.

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,102 @@
...
+[46985]: https://github.com/rust-lang/rust/pull/46985
+[47006]: https://github.com/rust-lang/rust/pull/47006
+[46830]: https://github.com/rust-lang/rust/pull/46830
+[46095]: https://github.com/rust-lang/rust/pull/46095
+[46666]: https://github.com/rust-lang/rust/pull/46666
+[cargo/5013]: https://github.com/rust-lang/cargo/pull/5013
+[cargo/5029]: https://github.com/rust-lang/cargo/pull/5029
+[RFC 1358]: https://github.com/rust-lang/rfcs/pull/1358
+[`Location::column`]: https://doc.rust-lang.org/std/panic/struct.Location.html#method.column
+[`ptr::NonNull`]: https://doc.rust-lang.org/std/ptr/struct.NonNull.html
```
Comment (JordiPolo) [inline RELEASES.md]: link returns 404?

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,102 @@
...
+- [`Duration::from_secs`][47300]
+- [`Duration::from_millis`][47300]
+- [`Duration::from_micros`][47300]
+- [`Duration::from_nanos`][47300]
+
+Cargo
+-----
+- [`cargo new` no longer removes `rust` or `rs` prefixs/suffixs.][cargo/5013]
+- [`cargo new` now defaults to creating a binary crate, instead of a
+  libary crate.][cargo/5029]
```
Comment (JordiPolo) [inline RELEASES.md]: libary -> library

File: RELEASES.md (hunk 4 above)
Comment (kennytm) [reply RELEASES.md]: The link should be available once 1.25 is released to stable channel.

**Comment by shepmaster** [comment]
Ping from triage, @Mark-Simulacrum — looks like it's back in your court!

File: RELEASES.md [hunk 6]
```diff
@@ -1,3 +1,102 @@
...
+- [Added `i586-unknown-linux-musl` target][47282]
+
+Libraries
+---------
+- [Impl Send for `process::Command` on Unix.][47760]
+- [Impl PartialEq and Eq for `ParseCharError`.][47790]
+- [`UnsafeCell::into_inner` is now safe.][47204]
+- [Implement libstd for CloudABI.][47268]
+- [`Float::{from_bits, to_bits}` is now available in libcore.][46931]
+
```
Comment (Mark-Simulacrum) [inline RELEASES.md]: Extra line?

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]

**Comment by Havvy** [comment]
Isn't this going to be the first release with Rust By Example in the distribution documentation? Shouldn't that be in the release notes?

**Comment by Aaronepower** [comment]
@Havvy Do you have the PR that does that so I can check if it's in beta?

**Comment by steveklabnik** [comment]
https://github.com/rust-lang/rust/pull/46196

but easier: https://doc.rust-lang.org/beta/rust-by-example/

The PR was marked relnotes...

**Comment by shepmaster** [comment]
Ping from triage, @Mark-Simulacrum — looks like it's back in your court!

**Comment by Mark-Simulacrum** [comment]
Looks like my two nits still stand (https://github.com/rust-lang/rust/pull/48374#pullrequestreview-102176354). Otherwise I think we are good to merge.

**Comment by Aaronepower** [comment]
@Mark-Simulacrum I've removed the extra line, and Misc is no longer empty so it shouldn't be removed.

**Review by Mark-Simulacrum — APPROVED** [review]

**Pull request merged**